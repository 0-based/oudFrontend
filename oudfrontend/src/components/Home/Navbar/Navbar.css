.fixed-top {
  position: fixed;
  top: 0;
  right: 0;
  left: 223px;
  z-index: 2;
  /* height: 68px; */
}
.fa-angle-left:before {
  margin-right: 20px;
}
.oud-nav {
  padding-top: 13px;
  padding-bottom: 13px;
}

.oud-btn {
  z-index: 2;
  border-style: solid;
  border-width: 2px;
  width: 150px;
  /* color: #f0a91a; */
  border: 2px solid #f0a91a;
  transition: all 0.4s linear;
  border-radius: 80px !important;
  border-radius: 0.85rem;
}
.oud-btn:hover {
  /* background-color: #fff; */
  border: 2px solid #ffffff;
  /* color: #f0a91a; */
  /* font-weight: 700; */
}
.icon_wrap {
  cursor: pointer;
}
.form-control {
  width: auto;
}
.search-input {
  color: #000;
  padding: 6px 20px;
  height: 40px;
  margin-right: 3px;
  line-height: 40px;
  width: 290px;
  border: 0;
  border-radius: 500px;
  text-overflow: ellipsis;
  outline: none;
}
input.empty {
  font-size: 18px;
  font-family: FontAwesome;
  font-style: normal;
  font-weight: normal;
  text-decoration: inherit;
}

.login-singup {
  margin-left: 10vw;
  overflow: hidden;
}
.left-arrow {
  /* font-weight: 600; */
  font-size: 1.7em;
}
.right-arrow {
  /* font-weight: 600; */
  font-size: 1.7em;
}
.left-arrow:hover {
  color: #f0a91a;
  cursor: pointer;
}
.right-arrow:hover {
  color: #f0a91a;
  cursor: pointer;
}
.signup-signin-link {
  color: #ffffff;
}

.login-signup-btn:hover a {
  color: #f0a91a;
}
@media (max-width: 480px) {
  .dropdown-me {
    left: 5px !important;
    top: 50px !important;
  }

  .search-input {
    width: 100%;
    margin-top: -50px;
    margin-bottom: 10px;
  }
  .navbar-toggler {
    width: 100% !important;
  }
  .form-inline {
    width: 100%;
  }
  .nav-back-forward {
    visibility: hidden;
    width: 100%;
    margin-left: 42% !important;
  }
}

@media (max-width: 993px) {
  .dropdown-me {
    left: 5px !important;
    top: 50px !important;
  }
  .oud-btn {
    width: 100%;
  }
  .signin {
    margin-bottom: 15px !important;
  }
  button.btn.oud-btn.my-2.my-sm-0.mr-3.login-signup-btn.signin {
    margin-bottom: 10px !important;
  }
  .profile {
    margin-top: 5px;
  }
}

.bg-darky {
  background: #000;
}

.profile {
  width: 40px;
  height: 40px;
  border-radius: 30%;
  /* border: 2px solid #494338; */
}

.username {
  font-size: 20px;
  margin-bottom: 0;
  font-weight: bold;
  margin-left: 7px;
  color: white;
}

.dropdown-me {
  left: -30px;
  top: 50px;
  background: #ca8e15;
}
.element {
  color: #fff;
  background: #ca8e15;
  margin-top: 0;
}
.element:hover {
  background: #dd9b18;
  color: #fff;
}
.upgrade {
  color: #fff;
}
.upgrade:hover a {
  color: #f0a91a;
}
