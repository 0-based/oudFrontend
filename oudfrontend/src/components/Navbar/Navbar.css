.fixed-top {
  position: fixed;
  top: 0;
  right: 0;
  left: 223px;
  z-index: 2;
}
.fa-angle-left:before {
  margin-right: 20px;
}

.oud-btn {
  color: #f0a91a;
  border: 2px solid #f0a91a;
  border-radius: 0.85rem;
  z-index: 3;
  border-style: solid;
  border-width: 2px;
  border-radius: 80px !important;
  transition: all 0.4s linear;
  width: 150px;
}
.oud-btn:hover {
  background-color: #f0a91a;
  color: #fff;
}
.icon_wrap {
  cursor: pointer;
}
.form-control {
  width: auto;
}
.search-input {
  color: #000;
  padding: 6px 20px;
  height: 40px;
  margin-right: 3px;
  line-height: 40px;
  width: 290px;
  border: 0;
  border-radius: 500px;
  text-overflow: ellipsis;
  outline: none;
}
input.empty {
  font-size: 18px;
  font-family: FontAwesome;
  font-style: normal;
  font-weight: normal;
  text-decoration: inherit;
}

.login-singup {
  margin-left: 10vw;
  overflow: hidden;
}
.left-arrow {
  font-weight: 600;
}
.right-arrow {
  font-weight: 600;
}
.left-arrow:hover {
  color: #f0a91a;
  cursor: pointer;
}
.right-arrow:hover {
  color: #f0a91a;
  cursor: pointer;
}
