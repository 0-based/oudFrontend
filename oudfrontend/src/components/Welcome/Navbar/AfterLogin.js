import React, { Component } from 'react'
import { Link } from 'react-router-dom';
import mask from './../../../assets/images/mask.png';
class AfterLogin extends Component {
  doLogOut = () => {
    localStorage.removeItem("accessToken");
  }
  render() {
    return (
      <li class="nav-item dropdown">
        <Link
          class="nav-link dropdown-toggle"
          id="navbarDropdown"
          role="button"
          data-toggle="dropdown"
          aria-haspopup="true"
          aria-expanded="false"
          to="/"
        >
          <img
            className="img-responsive"
            src={mask}
            alt="Profile Icon"
            data-testid="profImage"
          />
          <span>&nbsp;&nbsp;</span>Profile
                    </Link>
        <div class="dropdown-menu" aria-labelledby="navbarDropdown">
          <Link
            data-testid="Account"
            to="/overview"
            class="dropdown-item"
          >
            Account
                     </Link>
          <Link
            data-testid="LogOut"
            to="/signin"
            class="dropdown-item"
            onClick={() => this.doLogOut()}
          >
            Log Out
          </Link>
        </div>
      </li>
    )
  }
}

export default AfterLogin
