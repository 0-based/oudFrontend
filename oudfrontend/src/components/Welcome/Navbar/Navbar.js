import React, { Component } from 'react'

class Navbar extends Component {
  render() {
    return (
      <div>
        Navbar is Loaded !!
      </div>
    )
  }
}

export default Navbar
