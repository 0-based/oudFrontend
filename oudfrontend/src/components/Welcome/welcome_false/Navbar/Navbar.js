import React from 'react';
import { Link, Redirect } from 'react-router-dom';
// import NavbarRouter from '../../../routes/NavbarRouter';
import logo from '../../../../assets/images/Logoc.png';
import '../../welcome.css';
/**nav bar if the user is not logged in */
class Navbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
      redirect: false,
    };
  }
  setRedirect = () => {
    this.setState({
      redirect: true,
    });
  };
  toUserHomepage = () => {
    if (this.state.redirect) {
      return <Redirect to="/welcome-user" />;
    }
  };
  /**
   * set the logged in to true
   * @function
   * @returns {void}
   */
  doLogIn = () => this.setState({ loggedIn: true });
  /**
   * set the logged in to true
   * @function
   * @returns {void}
   */
  doLogOut = () => this.setState({ loggedIn: false });
  /**
   * just to see what is the status that in the real time
   * @function
   * @returns {boolean}
   */
  handleStatus = () =>
    this.props.status
      ? this.setState({ loggedIn: true })
      : this.setState({ loggedIn: false });
  /**
   * the render function that have the element inside
   * @function
   * @returns {JSX}
   */
  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-dark bg-custom NavBarStyle_bg-custom NavBarStyle_navbar-brand NavBarStyle_navbar-expand-lg">
          <Link to="/welcome-guest" className="navbar-brand">
            <img
              className="img-responsive oudLogo"
              src={logo}
              alt="Oud logo"
              data-testid="oudlogo"
            />
          </Link>
          <button
            data-testid="button"
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav ml-auto nav-items font-weight-bold NavBarStyle_nav-items">
              <li className="nav-item">
                <Link data-testid="Premium" to="/premium" className="nav-link">
                  Premium
                </Link>
              </li>
              <li className="nav-item">
                <Link data-testid="Help" to="/help" className="nav-link">
                  Help
                </Link>
              </li>
              <li className="nav-item">
                <Link
                  data-testid="Download"
                  to="/download"
                  className="nav-link"
                >
                  Download
                </Link>
              </li>
              <li className="block NavBarStyle_block nav-item">{'|'}</li>
              <li className="nav-item">
                <Link data-testid="SignUP" to="/signup" className="nav-link">
                  Sign up
                </Link>
              </li>
              {this.toUserHomepage()}
              <li className="nav-item">
                <Link
                  data-testid="Login"
                  onClick={this.setRedirect}
                  className="nav-link"
                  to="/signin"
                >
                  Log in
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    );
  }
}
export default Navbar;
