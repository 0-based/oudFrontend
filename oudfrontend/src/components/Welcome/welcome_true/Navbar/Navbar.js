import React, {Fragment} from 'react';
import {Link, NavLink} from 'react-router-dom';
// import NavbarRouter from '../../../routes/NavbarRouter';
import logo from '../../../../assets/images/Logo.png';
import mask from '../../../../assets/images/mask.png';
import '../../welcome.css';
/**nav bar if the user is logged in */
class Navbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
    };
  }
  /**
   * set the logged in to true
   * @function
   * @returns {void}
   */
  doLogIn = () => this.setState({loggedIn: true});
  /**
   * set the logged in to true
   * @function
   * @returns {void}
   */
  doLogOut = () => this.setState({loggedIn: false});
  /**
   * just to see what is the status that in the real time
   * @function
   * @returns {boolean}
   */
  handleStatus = () =>
    this.props.status
      ? this.setState({loggedIn: true})
      : this.setState({loggedIn: false});
  /**
   * the render function that have the element inside
   * @function
   * @returns {JSX}
   */
  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-dark bg-custom  NavBarStyle_navbar-expand-lg NavBarStyle_navbar-dark">
          <Link
            to="/welcomeUser"
            className="navbar-brand NavBarStyle_navbar-brand"
          >
            <img
              className="img-responsive oudLogo"
              src={logo}
              alt="Oud logo"
              data-testid="oudlogo"
              to="/welcomeUser"
            />
          </Link>
          <button
            data-testid="button"
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav ml-auto nav-items NavBarStyle_nav-items font-weight-bold">
              <li className="nav-item">
                <Link data-testid="Premium" to="/premium" className="nav-link">
                  Premium
                </Link>
              </li>
              <li data-testid="Help" className="nav-item">
                <Link to="/help" className="nav-link">
                  Help
                </Link>
              </li>
              <li className="nav-item">
                <Link
                  data-testid="Download"
                  to="/download"
                  className="nav-link"
                >
                  Download
                </Link>
              </li>
              <li className=" NavBarStyle_block nav-item">{'|'}</li>
              <li class="nav-item dropdown">
                <Link
                  class="nav-link dropdown-toggle"
                  id="navbarDropdown"
                  role="button"
                  data-toggle="dropdown"
                  aria-haspopup="true"
                  aria-expanded="false"
                  to="/"
                >
                  <img
                    className="img-responsive"
                    src={mask}
                    alt="Profile Icon"
                    data-testid="profImage"
                  />
                  <span>&nbsp;&nbsp;</span>Profile
                </Link>
                <div class="dropdown-menu" aria-labelledby="navbarDropdown">
                  <Link
                    data-testid="Account"
                    to="/overview"
                    class="dropdown-item"
                  >
                    Account
                  </Link>
                  <Link
                    data-testid="LogOut"
                    to="/welcomeGuest"
                    class="dropdown-item"
                    onClick={() => this.doLogOut()}
                  >
                    Log Out
                  </Link>
                </div>
              </li>
            </ul>
          </div>
        </nav>
        <div>{/* <NavbarRouter /> */}</div>
      </div>
    );
  }
}
export default Navbar;
