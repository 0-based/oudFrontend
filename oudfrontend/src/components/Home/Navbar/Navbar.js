import React, { Component } from "react";
import "./Navbar.css";
import { withRouter } from "react-router-dom";
import { BeforeLogin, AfterLogin } from "./BeforeAfterLogin"


class Navbar extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoggedIn: true
    }
  }

  handleClickOnSearch = () => {
    this.props.history.replace('search');
  }

  handleGoBack = () => {
    this.props.history.goBack();
  }

  handleGoForward = () => {
    this.props.history.goForward();
  }

  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-dark bg-darky fixed-top">
          <form className="form-inline">
            <div className="back-forward nav-back-forward">
              <div className="navbar-brand" onClick={this.handleGoBack}>
                <i className="fa fa-angle-left fa-2x left-arrow"></i>
              </div>
              <div className="navbar-brand" onClick={this.handleGoForward}>
                <i className="fa fa-angle-right fa-2x right-arrow"></i>
              </div>
            </div>
            <input
              type="search"
              className="search-input empty"
              id="iconified"
              placeholder="&#xF002; Search for Artists, Songs"
              aria-label="Search"
              onClick={this.handleClickOnSearch}
            />
          </form>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon toggler"></span>
          </button>

          <div className="collapse navbar-collapse login-signup" id="navbarSupportedContent">
            <ul className="navbar-nav mr-auto"></ul>
            <ul className="navbar-nav mr-auto"></ul>
            {
              (this.state.isLoggedIn) ? <AfterLogin /> : <BeforeLogin />
            }
          </div>
        </nav>
      </div >
    );
  }
}

export default withRouter(Navbar);

