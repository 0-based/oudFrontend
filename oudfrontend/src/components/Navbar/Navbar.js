import React, { Component } from "react";
import "./Navbar.css";
import { withRouter } from "react-router-dom";
import { BeforeLogin } from "./BeforeLogin/BeforeLogin"
import { AfterLogin } from "./AfterLogin/AfterLogin"

/**
 * Component to render all the stuff in Home page
 *
 *
 * @author Abdallah Hemdan
 *
 * @component
 *
 */

class Navbar extends Component {
  /**
   * @constructor
   *
   * @param {object} props - get props from higher components
   * @param {boolean} isLoggedIn - bool to indicate wether a use is logged in or not
   */

  constructor(props) {
    super(props)
    this.state = {
      /**
       * A variable to check if the user is logged or not to use it in enabling or disabling some data and navbar view
       * 
       * @property {boolean} 
       * 
       */
      isLoggedIn: this.props.isLoggedIn
    }
  }

  /**
   * A function to handle switch current route to search route on clicking in search bar
   *
   * @function
   *
   * @param {string} newRoute - new route to change to it
   * 
   * @returns {void} returns nothing, it just handle changing the routes
   */
  handleClickOnSearch = (newRoute) => {
    this.props.history.replace(`/${newRoute}`);
  }

  /**
   * A function to handle going back to last route we were used
   * 
   * @function
   * 
   * @returns {void} returns nothing, it just handle changing the routes to last one back
   * 
   */
  handleGoBack = () => {
    this.props.history.goBack();
  }

  /**
   * A function to handle going forward to the last route we were used
   * 
   * @function
   * 
   * @returns {void } returns nothing, it just handle changing the routes to last one forward
   */
  handleGoForward = () => {
    this.props.history.goForward();
  }

  componentDidMount() {
    if (this.props.isSearch) {
      this.nameInput.focus();
    }
  }
  /**
   * @function
   * @name render
   * @description Render all Navbar components
   * @returns {JSX} Component for Home
   */
  render() {
    return (
      <div>
        <nav
          className="navbar navbar-expand-lg navbar-dark bg-darky fixed-top oud-nav"
          data-testid="home-nav"
        >
          <form
            className="form-inline"
            data-testid="left-part"
          >
            <div
              className="back-forward nav-back-forward"
              data-testid="nav-back-forward"
            >
              <div
                className="navbar-brand"
                onClick={this.handleGoBack}
                data-testid="back-switch"
              >
                <i className="fa fa-angle-left fa-lg left-arrow"></i>
              </div>
              <div
                className="navbar-brand"
                onClick={this.handleGoForward}
                data-testid="forward-switch">
                <i className="fa fa-angle-right fa-lg right-arrow"></i>
              </div>
            </div>
            <input
              type="search"
              className="search-input empty"
              id="iconified"
              ref={(input) => { this.nameInput = input; }}
              placeholder="&#xF002; Search for Artists, Songs"
              aria-label="Search"
              onClick={() => this.handleClickOnSearch('search')}
              data-testid="search-input"
            />
          </form>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            data-testid="toggle-btn"
          >
            <span className="navbar-toggler-icon toggler"></span>
          </button>

          <div
            className="collapse navbar-collapse login-signup"
            id="navbarSupportedContent"
            data-testid="right-part"
          >
            <ul className="navbar-nav mr-auto"></ul>
            <ul className="navbar-nav mr-auto"></ul>
            {
              (this.state.isLoggedIn) ?
                <AfterLogin data-testid="right-after-login" /> :
                <BeforeLogin data-testid="right-before-login" />
            }
          </div>
        </nav>
      </div >
    );
  }
}


export default withRouter(Navbar);

