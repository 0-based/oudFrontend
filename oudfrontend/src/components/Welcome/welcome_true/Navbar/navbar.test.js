import React from 'react';
import { render } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './Navbar';
const setUP = (props = {}) => {
  const component = shallow(<Navbar />);
  return component;
};

describe('Body page testing ', () => {
  // eslint-disable-next-line no-unused-vars
  let component;
  beforeEach(() => {
    component = setUP();
  });
  it('render correctly component', () => {
    const TextInputComponent = shallow(<Navbar />);
    expect(TextInputComponent).toMatchSnapshot();
  });
  it('render a oud logo ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="oudlogo"]`);
    expect(logo.length).toBe(1);
  });
  it('render a  Button ', () => {
    const component = shallow(<Navbar />);
    const Button = component.find(`[data-testid="button"]`);
    expect(Button.length).toBe(1);
  });
  it('render a Premium ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="Premium"]`);
    expect(logo.length).toBe(1);
  });
  it('render a Help ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="Help"]`);
    expect(logo.length).toBe(1);
  });
  it('render a Download ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="Download"]`);
    expect(logo.length).toBe(1);
  });
  it('render a profImage ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="profImage"]`);
    expect(logo.length).toBe(1);
  });
  it('render a Account ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="Account"]`);
    expect(logo.length).toBe(1);
  });
  it('render a LogOut ', () => {
    const component = shallow(<Navbar />);
    const logo = component.find(`[data-testid="LogOut"]`);
    expect(logo.length).toBe(1);
  });
});
