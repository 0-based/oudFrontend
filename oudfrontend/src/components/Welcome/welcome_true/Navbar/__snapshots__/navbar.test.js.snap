// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body page testing  render correctly component 1`] = `ShallowWrapper {}`;
